app/utilities/supplementary_data_parser.py

```diff
@@ -67,25 +58,33 @@ class SupplementaryDataMetadataSchema(Schema, StripWhitespaceMixin):
 
     @validates_schema()
     def validate_dataset_and_survey_id(  # pylint: disable=unused-argument
-        self, data: Mapping, **kwargs: Any
+        self, payload: Mapping, **kwargs: Any
     ) -> None:
-        if data:
-            if data["dataset_id"] != self.context["dataset_id"]:
+        if payload:
+            if payload["dataset_id"] != self.context["dataset_id"]:
                 raise ValidationError(
                     "Supplementary data did not return the specified Dataset ID"
                 )
 
-            if data["survey_id"] != self.context["survey_id"]:
+            if payload["survey_id"] != self.context["survey_id"]:
                 raise ValidationError(
                     "Supplementary data did not return the specified Survey ID"
                 )
 
+            if (self.context["sds_schema_version"]) and (
+                payload["data"]["schema_version"] != self.context["sds_schema_version"]
```

Comment:
This could potentially lead to `KeyError` if the "schema_version" is not there, since we don't check payload[data] like we do check payload (`if payload:`). Is "schema_version" guaranteed to be in the payload[data]?